index.js

```diff
@@ -142,29 +144,45 @@ GeneratePackageJsonPlugin.prototype.apply = function(compiler) {
       }
     });
 
-    // Overwrite modules or set new module dependencies for those that have been
-    // deliberately set in " otherPackageValues.dependencies "
-    if (this.otherPackageValues && this.otherPackageValues.dependencies) {
-      const nonWebpackModuleNames = Object.keys(this.otherPackageValues.dependencies);
-
-      for (let k = 0; k < nonWebpackModuleNames.length; k += 1) {
-        const moduleName = nonWebpackModuleNames[k];
-
-        if (this.otherPackageValues.dependencies[moduleName] && this.otherPackageValues.dependencies[moduleName].length > 0) {
-          logIfDebug(`GPJWP: Adding deliberate module with version set deliberately: ${moduleName} -> ${this.otherPackageValues.dependencies[moduleName]}`);
-          modules[moduleName] = this.otherPackageValues.dependencies[moduleName];
-        } else if (this.dependencyVersionMap[moduleName] != null) {
-          logIfDebug(`GPJWP: Adding deliberate module with version found in sources: ${moduleName} -> ${this.dependencyVersionMap[moduleName]}`);
-          modules[moduleName] = this.dependencyVersionMap[moduleName];
-        } else {
-          console.warn(`GeneratePackageJsonPlugin: You have set a module to be included deliberately with name: "${moduleName}" - but there is no version specified in any source files!`);
+    const basePackageValues = Object.assign({}, this.otherPackageValues);
+
+    for (let i = 0; i < dependencyTypes.length; i += 1) {
+      const dependencyType = dependencyTypes[i];
```

Comment:
Not a big deal, but these lines could be reduced and made easier to read using the `for ... of` syntax:

```js
for (const dependencyType of dependencyTypes) {
  // ...
}
```